gnmi/gnmi.go

```diff
@@ -104,6 +107,82 @@ func (g *GNMI) storeMetric(ctx context.Context, md pmetric.Metrics) error {
 	return nil
 }
 
+func marshalHistogramDataPoint(p *pmetric.HistogramDataPoint) (*anypb.Any, error) {
+	hp := &ompb.HistogramDataPoint{
+		Count:          p.Count(),
+		BucketCounts:   p.BucketCounts().AsRaw(),
+		ExplicitBounds: p.ExplicitBounds().AsRaw(),
+	}
+
+	if p.HasSum() {
+		hp.Sum = proto.Float64(p.Sum())
+	}
+	if p.HasMin() {
+		hp.Min = proto.Float64(p.Min())
+	}
+	if p.HasMax() {
+		hp.Max = proto.Float64(p.Max())
+	}
+	any, err := anypb.New(hp)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Could this be a switch statement?

Reply:
It's not an "if else" with 3 cases, it's 3 different attributes that get copied separately iff. they are set.